iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp

```diff
@@ -308,9 +309,9 @@ class HALDispatchABI {
 
   Value getIndexValue(Location loc, int64_t value, OpBuilder &builder) {
     SmallVector<Value> folded;
-    builder.createOrFold<UnrealizedConversionCastOp>(
+    builder.createOrFold<LLVM::ConstantOp>(
         folded, loc, typeConverter->convertType(builder.getIndexType()),
-        builder.createOrFold<ConstantIndexOp>(loc, value));
+        builder.getI64IntegerAttr(value));
     return folded.front();
```

Comment:
errr... why does `getIndexValue` no longer return an index?

Reply:
Looking at the lit tests, the op is already returning an i64. Are we assuming all devices are i64 for index types. If so this should be fine

Reply:
we aren't - riscv32 isn't, for example - needs to be index

Reply:
discussed [on Discord](https://discord.com/channels/689900678990135345/690274711523164166/885943515304054824), seems fine, just looked weird